nvflare/dashboard/application/models.py

```diff
@@ -0,0 +1,105 @@
+# Copyright (c) 2022, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+from datetime import datetime
+
+from . import db
+
+
+class CommonMixin(object):
+    id = db.Column(db.Integer, primary_key=True)
+    name = db.Column(db.String(512), default="")
+    description = db.Column(db.String(512), default="")
+    created_at = db.Column(db.DateTime, nullable=False, default=datetime.utcnow)
+    updated_at = db.Column(db.DateTime, default=datetime.utcnow, onupdate=datetime.utcnow)
+
+    def asdict(self):
+        return {c.name: getattr(self, c.name) for c in self.__table__.columns}
+
+
+class Organization(CommonMixin, db.Model):
+    def asdict(self):
+        return {c.name: getattr(self, c.name) for c in self.__table__.columns if c.name in ("name",)}
+
+
+class Role(CommonMixin, db.Model):
+    pass
+
+
+class Capacity(db.Model):
+    id = db.Column(db.Integer, primary_key=True)
+    capacity = db.Column(db.String(1024), default="")
+    created_at = db.Column(db.DateTime, nullable=False, default=datetime.utcnow)
+    updated_at = db.Column(db.DateTime, default=datetime.utcnow, onupdate=datetime.utcnow)
+
+    def asdict(self):
+        return {c.name: getattr(self, c.name) for c in self.__table__.columns}
+
+
+class Project(db.Model):
+    id = db.Column(db.Integer, primary_key=True)
+    frozen = db.Column(db.Boolean, default=False)
+    short_name = db.Column(db.String(32), default="")
+    title = db.Column(db.String(128), default="")
+    description = db.Column(db.String(512), default="")
+    app_location = db.Column(db.String(64), default="")
+    ha_mode = db.Column(db.Boolean, default=False)
+    starting_date = db.Column(db.String(64), default="")
+    end_date = db.Column(db.String(64), default="")
+    overseer = db.Column(db.String(64), default="")
+    server1 = db.Column(db.String(64), default="")
+    server2 = db.Column(db.String(64), default="")
+    root_cert = db.Column(db.String(2048), default="")
```

Comment:
Could this server1 server2 be "servers"?
Or if we only have 2 servers change to primary_server, backup_server?

Reply:
We only support up to two servers for the project hosted by this dashboard instance.  I tried not to use primary or backup because those two servers are backup for each other.  When server1 is primary, server2 is the backup.  When server2 is primary, server1 is backup.